sync/notify/__init__.py

```diff
@@ -0,0 +1,48 @@
+import geckomsg
+import wptfyimsg
+
+
+def message_for_sync(sync):
+    wptfyi_parts = wptfyimsg.for_sync(sync)
+    gecko_parts = geckomsg.for_sync(sync)
+
+    msg_parts = []
+    if wptfyi_parts is not None:
+        msg_parts.extend(wptfyi_parts)
+    if gecko_parts is not None:
+        msg_parts.extend(gecko_parts)
+
+    if not msg_parts:
+        return
+
+    truncated, truncated_message = truncate_message(msg_parts)
+    if truncated:
+        message = "\n".join(msg_parts)
+    else:
+        message = truncated_message
+        truncated_message = None
+
+    return message, truncated_message
+
+
+def truncate_message(parts):
+    # This is currently the number of codepoints, with the proviso that
+    # only BMP characters are supported
+    MAX_LENGTH = 65535
+
+    suffix = "(See attachment for full changes)"
+    message = ""
+    truncated = False
+
+    padding = len(suffix) + 1
+    for part in parts:
+        if len(message) + len(part) + 1 > MAX_LENGTH - padding:
+            truncated = True
+            part = suffix
```

Comment:
Wouldn't it be better to do something like `part += '\n' + suffix` here? So that we maximise the amount of content in the comment

Reply:
Well no, because that would lead to an over-long comment. But we could do something like `part = part[:MAX_LENGTH - len(message) - padding] + '\n' + suffix`, but I'm not sure that having something cut off in the middle of a word is really better than just skipping the entire entry and hoping people look at the attachment.

Reply:
ahh yes sorry thats what I meant. I think either way is fine, I am just thinking that the more detail we can provide in the comment may lead to people not always having to download the attachment?

Reply:
Well hopefully having 65kB of text is pretty rare in the first place; that's a short novella worth of "important' failures to wade through.

Reply:
To sum up: you may be correct, but I don't think this should block landing this change. It's probably more useful to do followup work to condense the output than fiddle at the edges in terms of how we present too-long-for-a-comment data.

Reply:
Agreed!